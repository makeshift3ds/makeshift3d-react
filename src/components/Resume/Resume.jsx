import React from 'react';
import './Resume.css';

const Resume = () => (
  <div className="content-container">
    <article>
      <h1>Resume</h1>
      <p>This is about my professional career</p>
    </article>
    <aside>
      <div className="aside-block">
        <p>This guy codes!</p>
        <span> - Random Redditor</span>
      </div>
    </aside>
  </div>
);

export default Resume;
