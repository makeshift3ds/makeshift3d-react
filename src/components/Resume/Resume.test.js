import React from 'react';
import { shallow } from 'enzyme';
import Resume from './Resume.jsx';

describe('<Resume />', () => {
  it('should render 1 <Resume /> component', () => {
    const component = shallow(<Resume />);
    expect(component).toHaveLength(1);
  });

  it('should render an article element', () => {
    const component = shallow(<Resume />);
    expect(component.find('article')).toHaveLength(1);
  });

  it('should render an aside element', () => {
    const component = shallow(<Resume />);
    expect(component.find('aside')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<Resume />);
    expect(component).toMatchSnapshot();
  });
});
