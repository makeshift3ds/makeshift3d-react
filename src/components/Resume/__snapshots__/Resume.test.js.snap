// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resume /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Resume />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h1>
            Resume
          </h1>
          <p>
            This is about my professional career
          </p>
        </article>,
        <aside>
          <div
            className="aside-block"
          >
            <p>
              This guy codes!
            </p>
            <span>
               - Random Redditor
            </span>
          </div>
        </aside>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Resume
            </h1>,
            <p>
              This is about my professional career
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Resume",
            },
            "ref": null,
            "rendered": "Resume",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is about my professional career",
            },
            "ref": null,
            "rendered": "This is about my professional career",
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="aside-block"
          >
            <p>
              This guy codes!
            </p>
            <span>
               - Random Redditor
            </span>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                This guy codes!
              </p>,
              <span>
                 - Random Redditor
              </span>,
            ],
            "className": "aside-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This guy codes!",
              },
              "ref": null,
              "rendered": "This guy codes!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " - Random Redditor",
              },
              "ref": null,
              "rendered": " - Random Redditor",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "aside",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h1>
              Resume
            </h1>
            <p>
              This is about my professional career
            </p>
          </article>,
          <aside>
            <div
              className="aside-block"
            >
              <p>
                This guy codes!
              </p>
              <span>
                 - Random Redditor
              </span>
            </div>
          </aside>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Resume
              </h1>,
              <p>
                This is about my professional career
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Resume",
              },
              "ref": null,
              "rendered": "Resume",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is about my professional career",
              },
              "ref": null,
              "rendered": "This is about my professional career",
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="aside-block"
            >
              <p>
                This guy codes!
              </p>
              <span>
                 - Random Redditor
              </span>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  This guy codes!
                </p>,
                <span>
                   - Random Redditor
                </span>,
              ],
              "className": "aside-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This guy codes!",
                },
                "ref": null,
                "rendered": "This guy codes!",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " - Random Redditor",
                },
                "ref": null,
                "rendered": " - Random Redditor",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "aside",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
